javascript/javascript1/week4/js1week4.js

```diff
@@ -0,0 +1,91 @@
+//voice assistant
+//array for to do list
+const listToDo = [];
+//function that returns a response of a command
+let inputName='';
+function getReply(command){
+    //hello myname is Benjamin
+    let sentence = command.split(' ');
+    if(command=== 'Hello my name is Benjamin'){
+        if(inputName ===''){
+            inputName =sentence[sentence.length-1];
+            return('Nice to meet you '+inputName);
+        }
+        else{
+            return('Repeat sentence');
+        }
+    }
+        else if(command==='What is my name?') 
+        {
+          return('Your name is '+inputName);
+        }
+//Add fishing to my todo
+//Add singing in the shower to my todo
+ else if (sentence[0] ==='Add'){
+     listToDo.push(sentence.slice(1,sentence.indexOf('to')).join(' '));
+     return(sentence.slice(1,sentence.indexOf('to')).join(' ') + ' added to your todo');
+ }
+ //Remove fishing from todo
+ else if(sentence[0] ==='Remove'){
+     listToDo.splice(listToDo.indexOf(sentence.slice(1,sentence.indexOf('from')).join(' ')),1);
+     return('Removed '+sentence.slice(1,sentence.indexOf('from')).join(' ')+' from your todo');
+ }
+ else if(sentence[0] ==='What' && sentence[sentence.length-1] === 'todo?'){
+     return('You have '+listToDo.length+' todos - '+listToDo.join(' and '));
+ }
+ //what day is it today
+ else if(command =='What day is it today?')
+ {
+     let monthOfYear=['January','Febrauary','March','April','May','June','July','August','September','October','November','December'];
+     let month= new Date().getMonth();
+     month= monthOfYear[month];
+     return (new Date().getDate()+'.of '+ month +' '+new Date().getFullYear());
+ }
+ //set timer for 4 minutes
+ else if(command == 'Set a timer for 4 minutes')
+ {
+     let time = command.substr(16,2);
+     console.log('Timer set for '+time+' minutes');
+     let timeToMiliSeconds = time*60*1000;
+     setTimeout(function(){console.log('Timer Done');}, timeToMiliSeconds);
+ }
+     //Simple math
+     else{
+         return getmath(sentence);
+     }
+ }
+ function getmath(mathEquation){
+     const a = Number(mathEquation[0]);
+     const operator = mathEquation[1];
```

Comment:
` 9 + 4` is a string that can be represented as an array of chars ['9', ' ', '+',' ', '4']. therefore the operator is at index 2 not 1.
Do you agree ?

Reply:
Hi, Yes i can understand the logic behind the change. Space is also counted so the operator index value should be 2 instead of 1. But if i make the change as index 2 i get the result as undefined.

Reply:
Okay. You could try do do something like this `expressionArray=mathEquation.split(' ')`. And then I suppose what you have would always work.

Reply:
Okay. will try this.